app/models/visitors_step.rb

```diff
@@ -56,7 +59,9 @@ def visitors_attributes=(params)
 
   def valid?(*)
     # This must be eager because we want to show errors on all objects.
-    visitors.inject([super]) { |a, e| a << e.valid? }.all?
+    visitors_valid = visitors.map(&:valid?).all?
```

Comment:
Yes, that's a good point – I've amended to add a clear comment. Encouragingly the (lead visitor) test does fail if the order is inverted.